service/matching/matchingEngine.go

```diff
@@ -371,7 +371,7 @@ pollLoop:
 			})
 			if err != nil {
 				// will notify query client that the query task failed
-				e.deliverQueryResult(task.query.taskID, &queryResult{internalError: err}) // nolint:errcheck
+				_ = e.deliverQueryResult(task.query.taskID, &queryResult{internalError: err})
```

Comment:
why are we returning an error on deliverQueryResult if we're going to ignore it?

Reply:
this case, SDK is trying to get a new task, if server see error, server can basically tell SDK to retry (or give back an empty result)

below will use the error:
https://github.com/temporalio/temporal/blob/2dd1919a4444e9aaf512a10574976038762fc2fe/service/matching/matchingEngine.go#L541

Reply:
linter will complain